Tests/PackageLoadingTests/TargetSourcesBuilderTests.swift

```diff
@@ -148,6 +148,36 @@ class TargetSourcesBuilderTests: XCTestCase {
         }
     }
 
+    func testDoesNotErrorWithAdditionalFileRules() throws {
+        let target = try TargetDescription(
+            name: "Foo",
+            path: nil,
+            exclude: [],
+            sources: nil,
+            resources: [],
+            publicHeadersPath: nil,
+            type: .regular
+        )
+
+        let fs = InMemoryFileSystem()
+        fs.createEmptyFiles(at: .root, files: [
+            "/Foo.swift",
+            "/Bar.swift",
+            "/Baz.something/"
+        ])
+
+        let somethingRule = FileRuleDescription(
+            rule: .compile,
+            toolsVersion: .minimumRequired,
+            fileTypes: ["something"]
+        )
+
+        build(target: target, additionalFileRules: [somethingRule], toolsVersion: .v5, fs: fs) { _, _, _, _,_  in
+            // Does not throw.
```

Comment:
Might be good to check that the `target` contains the expected sources, e.g. this test would have passed before we fixed `additionalFileRules` regardless of the mixed sources check.

Reply:
Will do.